### PR TITLE
Tweak the dev setup a little

### PR DESCRIPTION
**Pull request by amazon-meaisiah**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: dev-portal/scripts/deploy-stack.js [hunk 1]
```diff
@@ -2,80 +2,23 @@
...
-
-function getRequired (key) {
-  const value = deployerConfig[key]
-  if (value) return value
-  errors.push(key + ' must be defined')
-}
-
-function getOptional (key, orElse) {
-  return deployerConfig[key] || orElse
+const deployTemplate = require('./deploy-template.js')
```
Comment (MynockSpit) [inline dev-portal/scripts/deploy-stack.js:9]: It's a little confusing that you call this `deployTemplate` when you use it, but it's defined as `deployerConfig` in the file.

File: dev-portal/scripts/deploy-template.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  // optional inputs
+  const staticAssetRebuildMode = getOptional('staticAssetRebuildMode', '')
+  const developmentMode = getOptional('developmentMode')
+
+  // AWS SAM CLI configuration
+  const awsSamCliProfile = getOptional('awsSamCliProfile')
+
+  if (missing.length) return missing
+
+  execute('sam', [
```
Comment (MynockSpit) [inline dev-portal/scripts/deploy-template.js:44]: Isn't `execute` async?

File: dev-portal/scripts/deploy-stack.js (hunk 1 above)
Comment (amazon-meaisiah) [reply dev-portal/scripts/deploy-stack.js:9]: `deployerConfig` is the argument - it's an arrow function. 😉

Also, the deployer config includes other things like profiles and such.

File: dev-portal/scripts/deploy-template.js (hunk 2 above)
Comment (amazon-meaisiah) [reply dev-portal/scripts/deploy-template.js:44]: [Nope](https://github.com/awslabs/aws-api-gateway-developer-portal/blob/staging/dev-portal/scripts/utils.js#L8)

File: dev-portal/scripts/deploy-stack.js (hunk 1 above)
Comment (MynockSpit) [reply dev-portal/scripts/deploy-stack.js:9]: So it is. I would much prefer it being a named function of some variety so people like me don't make that mistake and get confused again. :p

File: dev-portal/scripts/deploy-template.js (hunk 2 above)
Comment (MynockSpit) [reply dev-portal/scripts/deploy-template.js:44]: Ah, I was looking at my stale local version.

**Pull request merged**